src/modules/position-creator/position.creator.transaction.resolver.ts

```diff
@@ -1,108 +1,363 @@
 import { PositionCreatorTransactionService } from './services/position.creator.transaction';
 import { InputTokenModel } from 'src/models/inputToken.model';
 import { UserAuthResult } from '../auth/user.auth.result';
 import { AuthUser } from '../auth/auth.user';
+import {
+    DualFarmPositionSingleTokenModel,
+    EnergyPositionSingleTokenModel,
+    FarmPositionSingleTokenModel,
+    LiquidityPositionSingleTokenModel,
+    StakingPositionSingleTokenModel,
+} from './models/position.creator.model';
+import { PositionCreatorComputeService } from './services/position.creator.compute';
+import { FarmAbiServiceV2 } from '../farm/v2/services/farm.v2.abi.service';
+import { StakingProxyAbiService } from '../staking-proxy/services/staking.proxy.abi.service';
+import { GraphQLError } from 'graphql';
+import { ApolloServerErrorCode } from '@apollo/server/errors';
+import { constantsConfig } from 'src/config';
+import { StakingAbiService } from '../staking/services/staking.abi.service';
 
-@Resolver()
-@UseGuards(JwtOrNativeAuthGuard)
-export class PositionCreatorTransactionResolver {
+@Resolver(() => LiquidityPositionSingleTokenModel)
+export class LiquidityPositionSingleTokenResolver {
     constructor(
         private readonly posCreatorTransaction: PositionCreatorTransactionService,
     ) {}
 
-    @Query(() => TransactionModel)
-    async createPositionSingleToken(
+    @ResolveField(() => [TransactionModel])
+    @UseGuards(JwtOrNativeAuthGuard)
+    async transactions(
         @AuthUser() user: UserAuthResult,
-        @Args('pairAddress') pairAddress: string,
-        @Args('payment') payment: InputTokenModel,
-        @Args('tolerance') tolerance: number,
+        @Parent() parent: LiquidityPositionSingleTokenModel,
         @Args('lockEpochs', { nullable: true }) lockEpochs: number,
-    ): Promise<TransactionModel> {
-        return this.posCreatorTransaction.createLiquidityPositionSingleToken(
-            user.address,
-            pairAddress,
-            new EsdtTokenPayment({
-                tokenIdentifier: payment.tokenID,
-                tokenNonce: payment.nonce,
-                amount: payment.amount,
-            }),
-            tolerance,
-            lockEpochs,
-        );
+    ): Promise<TransactionModel[]> {
+        const pairAddress =
+            parent.swaps[parent.swaps.length - 1].pairs[0].address;
+        const payment = new EsdtTokenPayment({
+            tokenIdentifier: parent.swaps[0].tokenInID,
+            tokenNonce: 0,
+            amount: parent.swaps[0].amountIn,
+        });
+
+        const transactions =
+            await this.posCreatorTransaction.createLiquidityPositionSingleToken(
+                user.address,
+                pairAddress,
+                payment,
+                parent.swaps[parent.swaps.length - 1].tolerance,
+                parent.swaps,
+                lockEpochs,
+            );
+
+        return transactions;
     }
+}
 
-    @Query(() => [TransactionModel])
```

Comment:
I would extract a different class for each resolver

Reply:
they are different classes for each resolver 😄

Reply:
I wanted to say different files

Reply:
will refactor the module in another PR